From 5895f16defa560ad23ccfec344cdc3aaba9df34d Mon Sep 17 00:00:00 2001
From: Roger Siddons <dizygotheca@ntlworld.com>
Date: Thu, 9 Jul 2015 11:49:39 +0100
Subject: [PATCH 01/10] MediaMonitor: Support ejection of specific device

Mediamonitor only supports device ejection via its own dialog.

This patch allows MM clients to eject devices directly.

diff --git a/mythtv/libs/libmyth/mythmediamonitor.cpp b/mythtv/libs/libmyth/mythmediamonitor.cpp
index fc7d305..305bf74 100644
--- a/mythtv/libs/libmyth/mythmediamonitor.cpp
+++ b/mythtv/libs/libmyth/mythmediamonitor.cpp
@@ -240,6 +240,15 @@ void MediaMonitor::ChooseAndEjectMedia(void)
     AttemptEject(selected);
 }
 
+
+void MediaMonitor::EjectMedia(const QString &path)
+{
+    MythMediaDevice *device = GetMedia(path);
+    if (device)
+        AttemptEject(device);
+}
+
+
 void MediaMonitor::AttemptEject(MythMediaDevice *device)
 {
     QString  dev = DevName(device);
diff --git a/mythtv/libs/libmyth/mythmediamonitor.h b/mythtv/libs/libmyth/mythmediamonitor.h
index 6475f34..b9a3291 100644
--- a/mythtv/libs/libmyth/mythmediamonitor.h
+++ b/mythtv/libs/libmyth/mythmediamonitor.h
@@ -46,6 +46,7 @@ class MPUBLIC MediaMonitor : public QObject
     virtual void StartMonitoring(void);
     void StopMonitoring(void);
     void ChooseAndEjectMedia(void);
+    void EjectMedia(const QString &path);
 
     static MediaMonitor *GetMediaMonitor(void);
     static QString GetMountPath(const QString& devPath);
-- 
2.1.4

